builtin/logical/aws/client.go

```diff
@@ -73,6 +81,9 @@ func clientSTS(s logical.Storage) (*sts.STS, error) {
 		return nil, err
 	}
 	client := sts.New(session.New(awsConfig))
+	if *awsConfig.Endpoint != "none" {
+		client = sts.New(session.New(awsConfig.WithEndpoint(*awsConfig.Endpoint)))
```

Comment:
You're configuring STS and IAM clients with the same endpoints, and I doubt that will work. They likely need different endpoints.